### PR TITLE
Tell feature for chatops (resolves #68)

### LINKED ISSUE #68: `show` and `tell` from the cli
* `show` should publish a paste and post a link to the irc channel ( icarito@tycho posted thisfile.txt to http://paste... )
* `tell` should tell a specific user (or all) a message, from the cmd line

### PR DESCRIPTION
**Pull request by ryzokuken**
@icarito @jywarren @ananyo2012 How does this look?

### CODE REVIEW

**Comment by ryzokuken** [comment]
@jywarren @ananyo2012 yay!

**Comment by ryzokuken** [comment]
@icarito is this better?

File: spec/utils-spec.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const utils = require('../src/utils');
+
+describe('Utils Tests', () => {
```
Comment (ananyo2012) [inline spec/utils-spec.js]: Is this spec required now ?

**Comment by ananyo2012** [comment]
@ryzokuken Is this ready or WIP ?

File: spec/utils-spec.js (hunk 1 above)
Comment (ryzokuken) [inline spec/utils-spec.js]: yes it is.

**Comment by ryzokuken** [comment]
@ananyo2012 I could add `show` in this one itself or in another PR. This already seems big enough due to coverage increase. What do you think?

**Comment by ananyo2012** [comment]
@ryzokuken Yes sure you can make another PR for it. Can you rename the PR accordingly?

**Comment by ananyo2012** [comment]
@ryzokuken I am bit confused with the `tell` behavior. In the behavior spec I see a `speak` behavior been tested while in `bot.js` I see a case named `tell` . Can you explain the working a bit?

**Comment by ryzokuken** [comment]
@ananyo2012 the `tell` and `speak` behaviors are two completely different behaviors. The `speak` behavior is a real, simple (and somewhat useless) behavior that just returns back all the options. It is not really used in the bot, it's just used for testing purposes. On the other hand, `tell` is a real functionality, although it isn't a real conventional behavior. Basically, the express server is listening for action requests on `localhost`. In order to issue a `tell` action, you need to make an HTTP request internally. I have been working on a client (https://github.com/ryzokuken/plotsbot-client) to simplify this process on the command-line.

**Comment by ananyo2012** [comment]
Ok I get it now. So basically what it does is look for a `tell` and `message` param in the request and logs them on the console. I previously thought it of some behavior called by mentioning the bot like other behaviors. Can we expand on this a bit in the issue so it is clear?

Also we can go ahead with the tests for this in this same PR by having a mock Http client to test the tell behavior as well.

**Comment by ryzokuken** [comment]
@ananyo2012 @jywarren @icarito What's left for this to get merged? I should've done the testing improvements in another PR, that way that could have been merged quickly without affecting the rest of the changes in this one. 😅

**Comment by ryzokuken** [comment]
@icarito indeed, but I doubt all that added complexity has gone in vain. I mean, we have devised a way for local applications to communicate with the bot. We are certainly not using this functionality to its full functionality yet. Perhaps some different applications can be made to trigger different stuff on the chatbot side of things (eg: a cronjob that keeps looking if everything is okay every once a while and alerts everyone on the chatroom if there's some sort of issue).

**Comment by icarito** [comment]
It looks fine to me, if you want, I can try and test it, or we can merge this and rather work on making a jenkins job to auto-deploy in tycho.

**Comment by ryzokuken** [comment]
@icarito Sure. Seems good to me as well.

**Comment by jywarren** [comment]
If you think it's ready, go ahead!! :-)

**Comment by ryzokuken** [comment]
@icarito @jywarren so, should I merge this?

**Comment by jywarren** [comment]
yep!

**Comment by ryzokuken** [comment]
@icarito Feel free to check this out 🎉 

Also, let's work on the Jenkins job whenever you're free.

**Pull request merged**